packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/base/data_sources/remote/interceptors/auth_interceptor.dart

```diff
@@ -1,101 +1,113 @@
 {{> licence.dart }}
 
+import 'dart:async';
 import 'dart:developer';
 
 import 'package:dio/dio.dart';
-import 'package:provider/provider.dart';
 
+import '../../../common_use_cases/check_access_token_expiration_use_case.dart';
 import '../../../common_use_cases/fetch_access_token_use_case.dart';
 import '../../../common_use_cases/logout_use_case.dart';
+import '../../../common_use_cases/refresh_access_token_use_case.dart';
+import '../http_clients/api_http_client.dart';
 
-/// Interceptors are a simple way to intercept and modify http requests globally
-/// before they are sent to the server. That allows us to configure
-/// authentication tokens, add logs of the requests, add custom headers and
-/// much more.
-/// Interceptors can perform a variety of implicit tasks, from authentication
-/// to logging, for every HTTP request/response that is intercepted.
-///
-/// You should implement one or more methods from the contract.
-class AuthInterceptor extends Interceptor {
+class AuthInterceptor extends QueuedInterceptor {
   AuthInterceptor(
-    this._logoutUseCase,
+    this._apiHttpClient,
+    this._checkAccessTokenExpirationUseCase,
     this._fetchAccessTokenUseCase,
-    this._locator,
+    this._refreshAccessTokenUseCase,
+    this._logoutUseCase,
   );
 
-  final LogoutUseCase _logoutUseCase;
+  final ApiHttpClient _apiHttpClient;
+  final CheckAccessTokenExpirationUseCase _checkAccessTokenExpirationUseCase;
   final FetchAccessTokenUseCase _fetchAccessTokenUseCase;
-  final Locator _locator;
+  final RefreshAccessTokenUseCase _refreshAccessTokenUseCase;
+  final LogoutUseCase _logoutUseCase;
 
   @override
   Future<void> onRequest(
       RequestOptions options, RequestInterceptorHandler handler) async {
     final accessToken = await _fetchAccessTokenUseCase.execute();
+
+    // NOTE: Here you can check if the current token is expired or about to
+    // expire. In this case fetch a new token using `await` which will block
+    // the request queue until the token has been refreshed.
+    if (accessToken != null &&
+        _checkAccessTokenExpirationUseCase.execute(accessToken)) {
+      // try {
```

Comment:
I believe if we have a good default implementation the following code should be uncommented. Otherwise for each project the developers need to uncomment this solution in order to make it working, which potentially will lead to frustration.

Reply:
Implemented.